### PR TITLE
feat: implement MaxSupply

### PR DESCRIPTION
**Pull request by toteki**
## Description

This implements one of the token safety parameters on the leverage checklist.

closes: #1227

---

### Author Checklist

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### CODE REVIEW

**Comment by toteki** [comment]
Note: this will stay draft while we decide if `MaxSupply` will remain a flat amount, or be changed to a share of USD value.

File: x/leverage/fixtures/token.go [hunk 1]
```diff
@@ -31,6 +31,6 @@ func Token(base, symbol string) types.Token {
 		MaxCollateralShare:     sdk.MustNewDecFromStr("1"),
 		MaxSupplyUtilization:   sdk.MustNewDecFromStr("0.9"),
 		MinCollateralLiquidity: sdk.MustNewDecFromStr("0"),
-		MaxSupply:              sdk.NewInt(100000000000),
+		MaxSupply:              sdk.NewInt(100000_000000),
```
Comment (robert-zaremba) [inline x/leverage/fixtures/token.go:34]:
```suggestion
		MaxSupply:              sdk.NewInt(100_000_000000),
```

File: x/leverage/keeper/keeper_test.go [hunk 2]
```diff
@@ -28,6 +28,9 @@ func (s *IntegrationTestSuite) TestSupply() {
 	s.borrow(borrower, coin(atomDenom, 10_000000))
 	s.tk.SetBorrow(ctx, borrower, coin(atomDenom, 60_000000))
 
+	// create a supplier that will exceed token's default MaxSupply
+	whale := s.newAccount(coin(umeeDenom, 1000000_000000))
```
Comment (robert-zaremba) [inline x/leverage/keeper/keeper_test.go:32]:
```suggestion
	whale := s.newAccount(coin(umeeDenom, 1_000_000_000000))
```

Comment (robert-zaremba) [reply x/leverage/keeper/keeper_test.go:32]: or... let's define and export in fixtures a `DefaultMaxSupply=100_000000` and here let's use `fixtures.DefaultMaxSupply*10`.

File: x/leverage/keeper/keeper_test.go [hunk 3]
```diff
@@ -78,6 +81,13 @@ func (s *IntegrationTestSuite) TestSupply() {
 			coin("u/"+atomDenom, 40_000000),
 			nil,
 		},
+		{
+			"max supply",
+			whale,
+			coin(umeeDenom, 1000000_000000),
```
Comment (robert-zaremba) [inline x/leverage/keeper/keeper_test.go:87]: ditto, let's use `DefaultMaxSupply*5` for example

File: x/leverage/fixtures/token.go (hunk 1 above)
Comment (robert-zaremba) [reply x/leverage/fixtures/token.go:34]: Also, we decided to use share value.

**Review by robert-zaremba — CHANGES_REQUESTED** [review]
Let's use share value with a minimum total collateral to trigger the limitation (can be hardcoded in v3 as 500k I would say - in other words, the MaxSupply check will be triggered only if TotalAllSuply is >= 500k USD)
+ https://umeeworkspace.slack.com/archives/C03CVS0459A/p1662143001347019?thread_ts=1662045844.731089&cid=C03CVS0459A
+ https://github.com/umee-network/umee/issues/1227#issuecomment-1233167479

File: x/leverage/fixtures/token.go (hunk 1 above)
Comment (toteki) [reply x/leverage/fixtures/token.go:34]: We did not.

**Review by robert-zaremba — APPROVED** [review]
Following our discussion, we won't use that functionality. We can:
- leave this PR in draft
- or merge and add appropriate comments in the code
Do as you prefer.

**Pull request merged**